test/metabase/models/database_test.clj

```diff
@@ -216,6 +216,23 @@
                   (is (nil? (db/select-one Secret :id secret-id))
                       (format "Secret ID %d was not removed from the app DB" secret-id)))))))))))
 
+(deftest user-may-not-update-sample-database-test
+  (mt/with-temp Database [{:keys [id details] :as sample-database} {:engine    :h2
+                                                                    :is_sample true
+                                                                    :name      "Sample Database"
+                                                                    :details   {:db "./resources/sample-database.db;USER=GUEST;PASSWORD=guest"}}]
+    (testing " updating the engine of a sample database is not allowed"
+      (try (db/update! Database id :engine :sqlite)
+           (catch Exception e
+             (is (= "The engine on a sample database cannot be changed." (.getMessage e)))
+             (is (= {:status-code     400
+                     :existing-engine :h2
+                     :new-engine      :sqlite}
+                    (ex-data e))))))
```

Comment:
You should do `(is (thrown-with-msg? ...))` here because if this **does not** throw an Exception then those `is` forms inside the `catch` are never going to run. `thrown?` and `throw-with-msg?` actually assert that an Exception is thrown and fail if one is not